superset-frontend/src/components/Table/cell-renderers/ActionCell/ActionCell.test.tsx

```diff
@@ -0,0 +1,49 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import { render, screen } from 'spec/helpers/testing-library';
+import { configure } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import ActionCell, { appendDataToMenu } from './index';
+import { exampleMenuOptions, exampleRow } from './fixtures';
+
+test('renders with default props', async () => {
+  configure({ testIdAttribute: 'data-test' });
+  const clickHandler = jest.fn();
+  exampleMenuOptions[0].onClick = clickHandler;
+  render(<ActionCell menuOptions={exampleMenuOptions} row={exampleRow} />);
+  // Open the menu
+  await userEvent.click(screen.getByTestId('dropdown-trigger'));
+  // verify all of the menu items are being displayed
+  exampleMenuOptions.forEach((item, index) => {
+    expect(screen.getByText(item.label)).toBeInTheDocument();
+    if (index === 0) {
+      // verify the menu items' onClick gets invoked
+      userEvent.click(screen.getByText(item.label));
+    }
+  });
+  expect(clickHandler).toHaveBeenCalled();
+});
+
+test('appendDataToMenu utility', () => {
+  exampleMenuOptions.forEach(item => expect(item?.row).toBeUndefined());
+  const modifiedMenuOptions = appendDataToMenu(exampleMenuOptions, exampleRow);
+  modifiedMenuOptions.forEach(item => expect(item?.row).toBeDefined());
+});
```

Comment:
@lyndsiWilliams this unit test verifies that the `appendDataToMenu` utility function works as expected.  This is a way of testing code used within components without having to invoke the entire component lifecycle to see if a function behaves correctly.  Do you have any opinions on what could be added to avoid confusion / make this more clear? thanks!

Reply:
Oh I see, thank you for the explanation! Maybe just a comment in the test explaining what it's doing just in case 😁

Reply:
👍